src/traces/scattermapbox/hover.js

```diff
@@ -0,0 +1,91 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+var Fx = require('../../plots/cartesian/graph_interact');
+var getTraceColor = require('../scatter/get_trace_color');
+
+
+module.exports = function hoverPoints(pointData, xval, yval) {
+    var cd = pointData.cd,
+        trace = cd[0].trace,
+        xa = pointData.xa,
+        ya = pointData.ya;
+
+    // compute winding number about [-180, 180] globe
+    var winding = (xval >= 0) ?
```

Comment:
`mapbox -gl` maps repeat themselves on tiny zoom levels:

![image](https://cloud.githubusercontent.com/assets/6675409/15981100/fb21b654-2f3e-11e6-97ca-db15b28cda53.png)

So we need to handle the cases where `x` in data coords in beyond the [-180, 180] interval.